### DISCUSSION TITLE
Replacing number value in template

### DISCUSSION THREAD

**Discussion by Tobi0322** [hairyhenderson/gomplate#1187] [Q&A] [answered]
Hey Hairyhenderson,

I really like your tool. Thanks for sharing this. 

I have an AWS ECS task definition which I template using gomplate. String replacement works fine, but I am having issues to find a solution to replace numbers in my .json file (e.g. the CPU size). Considering the following JSON file where AWS_ECS_TASK_CPU is a number.
```
{
     ...,
     "cpu": "AWS_CPU",
     ...
}
```

How do I replace the CPU value in the JSON? I tried the following approach. But it says that replaceAll needs a String. I checked the documentation but did not find any proper solution. 
```
{{$file := (datasource "template") | data.ToJSON }}
{{$fileReplaceCPU := $file |  replaceAll "AWS_CPU" (conv.ToInt (datasource "config").AWS_CPU) }}
```
Thanks in advance,
Tobi

**Comment by hairyhenderson** [comment, marked as answer]
@Tobi0322 It looks like you're trying to use the datasource as a template. Is there a reason not to simply make it the template?

Something like:

```
{
     ...,
     "cpu": {{ (ds "config").AWS_CPU }},
     ...
}
```

If that doesn't make sense, maybe you could provide more context (the full command, as well as all datasources)...

**Reply by Tobi0322** [reply]
I see. I think I used the package the wrong way. This actually makes a lot more sense. 
Thank you for the quick reply. 
Greetings from Munich,
Tobi
